packages/app/src/cli/models/app/app.ts

```diff
@@ -20,13 +20,215 @@ export const LegacyAppSchema = zod
-const validateUrl = (zodType: zod.ZodString) => {
+const validateUrl = (zodType: zod.ZodString, {httpsOnly = false, message = 'Invalid url'} = {}) => {
+  const regex = httpsOnly ? /^(https:\/\/)/ : /^(https?:\/\/)/
   return zodType
     .url()
-    .refine((value) => Boolean(value.match(/^(https?:\/\/)/)), {message: 'Invalid url'})
-    .refine((value) => !value.includes('\n'), {message: 'Invalid url'})
+    .refine((value) => Boolean(value.match(regex)), {message})
+    .refine((value) => !value.includes('\n'), {message})
 }
 
+const ensurePathStartsWithSlash = (arg: unknown) => (typeof arg === 'string' && !arg.startsWith('/') ? `/${arg}` : arg)
+const ensureHttpsOnlyUrl = validateUrl(zod.string(), {
+  httpsOnly: true,
+  message: 'Only https urls are allowed',
+})
+
+const SubscriptionEndpointUrlValidation = ensureHttpsOnlyUrl.optional()
+const PubSubProjectValidation = zod.string().optional()
+const PubSubTopicValidation = zod.string().optional()
+const ArnValidation = zod
+  .string()
+  .regex(
+    /^arn:aws:events:(?<aws_region>[a-z]{2}-[a-z]+-[0-9]+)::event-source\/aws\.partner\/shopify\.com(\.test)?\/(?<api_client_id>\d+)\/(?<event_source_name>.+)$/,
+  )
+  .optional()
+
+const WebhookSubscriptionSchema = zod.object({
+  topic: zod.string(),
+  sub_topic: zod.string().optional(),
+  format: zod.enum(['json', 'xml']).optional(),
+  include_fields: zod.array(zod.string()).optional(),
+  metafield_namespaces: zod.array(zod.string()).optional(),
+  subscription_endpoint_url: SubscriptionEndpointUrlValidation,
+  path: zod
+    .string()
+    .refine((path) => path.startsWith('/') && path.length > 1, {
+      message: 'Path must start with a forward slash and be longer than 1 character',
+    })
+    .optional(),
+  pubsub_project: PubSubProjectValidation,
+  pubsub_topic: PubSubTopicValidation,
+  arn: ArnValidation,
+})
+
+const WebhooksSchema = zod
+  .object({
+    api_version: zod.string(),
+    privacy_compliance: zod
+      .object({
+        customer_deletion_url: ensureHttpsOnlyUrl.optional(),
+        customer_data_request_url: ensureHttpsOnlyUrl.optional(),
+        shop_deletion_url: ensureHttpsOnlyUrl.optional(),
+      })
+      .optional(),
+    subscription_endpoint_url: SubscriptionEndpointUrlValidation,
+    pubsub_project: PubSubProjectValidation,
+    pubsub_topic: PubSubTopicValidation,
+    arn: ArnValidation,
+    topics: zod.array(zod.string()).optional(),
+    subscriptions: zod.array(WebhookSubscriptionSchema).optional(),
+  })
+  .superRefine(
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    ({subscription_endpoint_url, pubsub_project, pubsub_topic, arn, topics = [], subscriptions = []}, ctx) => {
+      const topLevelDestinations = [subscription_endpoint_url, pubsub_project && pubsub_topic, arn].filter(Boolean)
+      const pubSubValidationErrMsg = 'You must declare both pubsub_project and pubsub_topic if you wish to use'
```

Comment:
It gets interpolated below in the top level and sub level validations:

<img width="874" alt="image" src="https://github.com/Shopify/cli/assets/29992628/baa0722f-bd7b-479f-9279-b70a43d6d6f8">


but that can be confusing haha, turning this into a function might make things a little more clearer. This also goes against everything I know about i18n, but I believe CLI is english only, so this might be alright

Reply:
Also side note is I'll need Jen to take a look at all my content once things are cleaned up. I was winging most of these!